source/D2Lang/src/D2Unicode/D2UnicodeStr.cpp

```diff
@@ -117,6 +117,25 @@ int __fastcall Unicode::strlen(const Unicode* str) {
   return i;
 }
 
+Unicode* __fastcall Unicode::strncat(
+    Unicode* dest,
+    const Unicode* src,
+    int count) {
+  int i_dest = 0;
+  while (dest[i_dest].ch != L'\0') {
+    ++i_dest;
+  }
+
+  int i_src;
+  for (i_src = 0; i_src != count && src[i_src].ch != L'\0'; ++i_src) {
```

Comment:
Similarly to `strcat`, should we implement this in terms of `strncpy` ?

Reply:
`Unicode::strncpy` and C99 standard conformant `strncpy` has the additional effect of adding null-terminating characters if all of the characters from the source is reached, but count characters have not been written.  `strncat` does not observe this behavior.